ojmicroline_thermostat/models/thermostat.py

```diff
@@ -116,24 +172,51 @@ def get_target_temperature(self) -> int:
         Returns:
             The current temperature.
         """
-        return self.temperatures[self.regulation_mode]
+        # The WG4 API makes this easy:
+        if self.set_point_temperature is not None:
+            return self.set_point_temperature
+
+        # The OWD5 API requires computing the value:
+        if self.regulation_mode == REGULATION_SCHEDULE and self.schedule:
+            return Schedule.from_json(self.schedule).get_active_temperature()
```

Comment:
Note that the schedule is now lazily parsed in `get_target_temperature`, and only if needed by the current regulation mode.

I'm open to changing it, but this option seemed simplest.